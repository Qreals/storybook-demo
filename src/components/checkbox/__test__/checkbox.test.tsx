import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Checkbox from '../index';
import mountTest from '../../../utils/mount-test';

describe('Checkbox', () => {
	mountTest('Checkbox mount unmount', Checkbox);

	test('should match snapshot', () => {
		const { container } = render(<Checkbox />);
		expect(container).toMatchSnapshot();
	});

	test('it should render with checked state', () => {
		render(<Checkbox checked={true} />);
		expect(screen.getByRole('checkbox')).toBeChecked();
	});

	test('it should render with disabled state', () => {
		render(<Checkbox disabled checked={true} />);
		expect(screen.getByRole('checkbox')).toBeDisabled();
	});

	test('it should trigger onChange event', () => {
		const onChange = jest.fn();
		render(<Checkbox onChange={onChange} />);
		const element = screen.getByRole('checkbox');
		fireEvent.click(element);
		expect(onChange).toHaveBeenCalledTimes(1);
	});
});
