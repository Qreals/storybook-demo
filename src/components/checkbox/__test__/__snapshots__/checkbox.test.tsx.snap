// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<div>
  <div
    class="gc-checkbox inline-block undefined"
  >
    <label
      class="ant-checkbox-wrapper flex items-start"
    >
      <span
        class="ant-checkbox"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value=""
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
    </label>
  </div>
</div>
`;
